pkg/web_app/lib/src/account.dart

```diff
@@ -159,38 +159,56 @@ class _AccountNavWidget {
 }
 
 class _PkgAdminWidget {
-  Element _root;
-  void init(Element root) {
+  Element _unauthenticatedRoot;
+  Element _unauthorizedRoot;
+  Element _authorizedRoot;
+
+  void init() {
     if (!pageData.isPackagePage) return;
-    _root = root;
-    _root.append(elem('h3', classes: ['title'], text: 'Admin'));
-    final discontinueToggle = elem(
-      'a',
-      text: pageData.pkgData.isDiscontinued
-          ? 'Remove "discontinued"'
-          : 'Mark as "discontinued"',
-      onClick: (_) async {
-        final options =
-            PkgOptions(isDiscontinued: !pageData.pkgData.isDiscontinued);
-        final rs = await client.put(
-          '/api/packages/${pageData.pkgData.package}/options',
-          body: json.encode(options.toJson()),
-        );
-        final map = json.decode(rs.body) as Map<String, dynamic>;
-        if (rs.statusCode == 200) {
-          window.location.reload();
-        } else {
-          window.alert(map['error'] as String);
-        }
-      },
-    );
-    _root.append(elem('p', children: [discontinueToggle]));
+    _unauthenticatedRoot = document.getElementById('-admin-unauthenticated');
+    _unauthorizedRoot = document.getElementById('-admin-unauthorized');
+    _authorizedRoot = document.getElementById('-admin-authorized');
     update();
   }
 
+  Future _toogleDiscontinued() async {
+    if (!window.confirm(
+        'Are you sure you want change the "discontinued" status of the package?')) {
+      return;
+    }
+    final options =
+        PkgOptions(isDiscontinued: !pageData.pkgData.isDiscontinued);
+    final rs = await client.put(
+      '/api/packages/${pageData.pkgData.package}/options',
+      body: json.encode(options.toJson()),
```

Comment:
Let's not do this now :)

But would be nice if in `package:client_data` we had something like:
```dart
class PkgOptions {
  ...
  fromJson(...) => ...
  toJson() {...};
}

class PubApiException implements Exception {
  final int status:
  final String mesage;
  ...
}

class PubApi {
  final http.Client _client;
  PubApi(this._client);

  Future<void> putPkgOptions(PkgOptions options) async {
    // HTTP + JSON related boiler plate only.
  }
}
```

Reply:
Good idea. We could probably reuse that in the pub-integration tests too.